src/backend/api/meals.js

```diff
@@ -1,12 +1,109 @@
+const { request, response } = require("express");
 const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+// Returns all meals
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    //   // knex syntax for selecting things. Look up the documentation for knex for further info
+    //   const titles = await knex("Meal").select("*");
+    //   response.json(titles);
+
+    const { maxPrice, availableReservations, title, createdAfter, limit } =
+      request.query;
+
+    let meals = knex("Meal");
+    //Get meals that has a price smaller than maxPrice
+    if (maxPrice) {
+      meals.where("price", "<", maxPrice);
+    }
+
+    // Get meals that still has available reservations
+    if (availableReservations) {
+      meals
+        .join("Reservation", "Reservation.meal_id", "=", "Meal.id")
+        .where("max_reservation", ">", "number_of_guests");
```

Comment:
available reservations need more specificity and that is way complicated at this stage. Try to discuss that with any of your teachers/mentors for more clarity.